docs/components/configMaps.md

```diff
@@ -0,0 +1,146 @@
+See the AUTHORS.md file at the top-level directory of this distribution and at
+https://github.com/fluid-lab/gamepad-navigator/raw/master/AUTHORS.md.
+
+Licensed under the BSD 3-Clause License. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the BSD 3-Clause License at
+https://github.com/fluid-lab/gamepad-navigator/blob/master/LICENSE
+-->
+
+# `gamepad.configMaps`
+
+This grade provides the configuration maps for the gamepad in the form of component's model data.
+
+## Using this grade
+
+This can be used by passing it to the component's `gradeNames` option.
+
+``` javascript
+fluid.defaults("my.configMaps.grade", {
+    gradeNames: ["gamepad.configMaps"]
+};
+
+var configMapsInstance = my.configMaps.grade();
+```
+
+## Component Options
+
+This grade doesn't use or provide any configuration options. It only uses the model data to provide gamepad configuration
+maps to other components. However, the grade can be extended to provide custom configuration options.
+
+## Model Structure
+
+The grade stores the model data in the following format:
+
+``` javascript
+model: {
+    map: {
+        buttons: {
+            "0": {
+                defaultAction: "previousPageInHistory",
+                currentAction: null,
+                speedFactor: 1,
+                background: false
+            },
+            // The buttons below use same format as above but have different values for configuration options (keys).
+            "1": {...},
+            "2": {...},
+            "3": {...},
+            "4": {...},
+            "5": {...},
+            "6": {...},
+            "7": {...},
+            "8": {...},
+            "9": {...},
+            "10": {...},
+            "11": {...},
+            "12": {...},
+            "13": {...},
+            "14": {...},
+            "15": {...},
+            // The buttons below are reserved for special functionality so they cannot be configured.
+            "16": null,
+            "17": null
+        },
+        axes: {
+            "0": {
+                defaultAction: "scrollHorizontally",
+                currentAction: null,
+                speedFactor: 1,
+                invert: false
+            },
+            // The axes below use the same format as above but have different values for configuration options (keys).
+            "1": {...},
+            "2": {...},
+            "3": {...}
+        }
+    },
+    // Stores all the common configuration options for the gamepad actions.
+    commonConfiguration: {
+        homepageURL: "https://www.google.com/"
+    }
+}
+```
+
+The first 16 buttons and all the axes are configurable. The configuration options used in their model data are explained
+below:
+
+| Option | Supported Input | Description |
+| :---: | :---: | :--- |
+| `defaultAction` | `buttons` and `axes` | The default action triggered with the particular gamepad input if the `currentAction` is not provided. |
```

Comment:
You should add a section on supported actions.  This can be another page.

Reply:
I thought of adding the controls to the README file. But I can add it here or as a separate page if you prefer.

Reply:
If you're going to document the actions in a separate file/ticket, that's fine, then just file an issue and leave a TODO here.